aspnetcore/fundamentals/dependency-injection.md

```diff
@@ -158,69 +179,85 @@ The `Startup.ConfigureServices` method is responsible for defining the services
+The following code is an example of how to add additional services to the container using the extension methods [AddDbContext\<TContext>](/dotnet/api/microsoft.extensions.dependencyinjection.entityframeworkservicecollectionextensions.adddbcontext) and <xref:Microsoft.Extensions.DependencyInjection.IdentityServiceCollectionExtensions.AddIdentityCore*>:
 
-    services.AddIdentity<ApplicationUser, IdentityRole>()
-        .AddEntityFrameworkStores<ApplicationDbContext>()
-        .AddDefaultTokenProviders();
+The following `ConfigureServices` is generated by the Razor Pages template using individual user accounts:
 
-    ...
-}
-```
+[!code-csharp[](dependency-injection/samples/3.x/DependencyInjectionSample/StartupEF.cs?name=snippet)]
 
-For more information, see the <xref:Microsoft.Extensions.DependencyInjection.ServiceCollection> class in the API documentation.
+For more information, <xref:Microsoft.Extensions.DependencyInjection.ServiceCollection> and <xref:security/authentication/identity>.
+
+See the section [Combining service collection](#csc) for instructions on writing an extension method to register services.
 
 ## Service lifetimes
 
 Choose an appropriate lifetime for each registered service. ASP.NET Core services can be configured with the following lifetimes:
 
 ### Transient
 
-Transient lifetime services (<xref:Microsoft.Extensions.DependencyInjection.ServiceCollectionServiceExtensions.AddTransient*>) are created each time they're requested from the service container. This lifetime works best for lightweight, stateless services.
+Transient lifetime services (<xref:Microsoft.Extensions.DependencyInjection.ServiceCollectionServiceExtensions.AddTransient*>) are created each time they're requested from the service container. This lifetime works best for lightweight, stateless services. Transient services don't need to be thread safe.
 
 In apps that process requests, transient services are disposed at the end of the request.
 
 ### Scoped
 
 Scoped lifetime services (<xref:Microsoft.Extensions.DependencyInjection.ServiceCollectionServiceExtensions.AddScoped*>) are created once per client request (connection).
 
+When using Entity Framework Core, the [AddDbContext](/dotnet/api/microsoft.extensions.dependencyinjection.entityframeworkservicecollectionextensions.adddbcontext) extension method registers `DbContext` types with a scoped lifetime by default.
+
 In apps that process requests, scoped services are disposed at the end of the request.
 
-> [!WARNING]
-> When using a scoped service in a middleware, inject the service into the `Invoke` or `InvokeAsync` method. Don't inject via [constructor injection](xref:mvc/controllers/dependency-injection#constructor-injection) because it forces the service to behave like a singleton. For more information, see <xref:fundamentals/middleware/write#per-request-middleware-dependencies>.
+Use scoped services in a middleware with one of the following approaches:
```

Comment:
@rahulpnath please review this change. I'm rewriting the entire doc and should be finished in a couple days. WHen I'm done, it would be great if you could review the updates. I can post a link to the new doc build.
I dropped the warning because this is not a trap you can fall into. The runtime throws and exception if you try to do this.

Reply:
Looks good and better now!